src/python/pants/util/tarutil.py

```diff
@@ -6,77 +6,77 @@
                         unicode_literals, with_statement)
 
 import tarfile
+from builtins import str
 
-import six
+from future.utils import implements_iterator
 
 
-if six.PY2:
-  class TarFile(tarfile.TarFile):
-    def next(self):
-      """A copy and modification of the next() method in tarfile module.
+@implements_iterator
```

Comment:
Py2 expects the function `next()` whereas Py3 expects `__next__()`. This decorator handles that discrepancy.